src/atlas/projection/detail/CubedSphereEquiAnglProjection.cc

```diff
@@ -0,0 +1,103 @@
+/*
+ * (C) Copyright 2020 UCAR
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+#include "CubedSphereEquiAnglProjection.h"
+
+#include <cmath>
+#include <iostream>
+
+#include "eckit/config/Parametrisation.h"
+#include "eckit/types/FloatCompare.h"
+#include "eckit/utils/Hash.h"
+
+#include "atlas/projection/detail/ProjectionFactory.h"
+#include "atlas/projection/detail/ProjectionUtilities.h"
+#include "atlas/runtime/Exception.h"
+#include "atlas/util/Config.h"
+#include "atlas/util/Constants.h"
+#include "atlas/util/CoordinateEnums.h"
+#include "atlas/util/Earth.h"
+
+namespace atlas {
+namespace projection {
+namespace detail {
+
+// -------------------------------------------------------------------------------------------------
+
+CubedSphereEquiAnglProjection::CubedSphereEquiAnglProjection( const eckit::Parametrisation& params )
+                                                               : CubedSphereProjectionBase(params) {
+  // Get Base data
+  const auto cubeNx = getCubeNx();
+  auto tile1Lats = getLatArray();
+  auto tile1Lons = getLonArray();
+
```

Comment:
Is there some example where it would be helpful to not have auto? To my mind this makes the code a cleaner and I'm not sure why it matters that it's hidden. Not opposed to changing I just want to make sure I understand.